### PR TITLE
ir: added CurlyBrace flag to ArrayDimFetchExpr, which indicates whether the brace is a curly brace

### PR DESCRIPTION
**Pull request by i582**

### CODE REVIEW

File: src/ir/irconv/irconv_test.go [hunk 1]
```diff
@@ -313,3 +318,59 @@ func TestInterpretString(t *testing.T) {
...
+			curlyBrace: true,
+		},
+		{
+			code:       `$arr['x'   ];`,
+			curlyBrace: false,
+		},
+	}
+
+	for _, test := range tests {
+		p := php7.NewParser([]byte(`<?php ` + test.code))
```
Comment (quasilyte) [inline src/ir/irconv/irconv_test.go:362]: Is it possible to use parseutil function for parsing here?
It's painful to use the parser directly in different places as we're going to migrate soon.

**Review by quasilyte — CHANGES_REQUESTED** [review]
Please add some phpgrep tests and update the matcher code so it handles this flag correctly.

**Comment by i582** [comment]
> Please add some phpgrep tests and update the matcher code so it handles this flag correctly.

Maybe in the next PR?

**Comment by i582** [comment]
There are not many changes, so let it be in this PR.

File: src/phpgrep/matcher_test.go [hunk 2]
```diff
@@ -422,6 +422,8 @@ func TestMatch(t *testing.T) {
 		{`$x->ff(1, 2)`, `$this->ff(1, 2)`},
 
 		{`$_[0]`, `$v[0]`},
+		{`$_{0}`, `$v{0}`},
+		{`$_{0}`, `$v[0]{0}`},
```
Comment (quasilyte) [inline src/phpgrep/matcher_test.go:426]: You also need to add negative tests.
For example, `$_[$_]` should not match `$_{$_}` and vice versa.

**Pull request merged**